components/operations-controller/internal/webhook/client.go

```diff
@@ -207,7 +221,13 @@ func parseResponseObject(resp *http.Response) (*web_hook.ResponseObject, error)
 func checkForErr(resp *http.Response, successStatusCode *int, error *string) error {
 	var errMsg string
 	if *successStatusCode != resp.StatusCode {
-		errMsg += fmt.Sprintf("response success status code was not met - expected %q, got %q; ", *successStatusCode, resp.StatusCode)
+		respBody, err := ioutil.ReadAll(resp.Body)
```

Comment:
reading the resp body twice in the same method Do/Poll should really be avoided. Instead of adding workarounds such as resp.Body = ioutil.NopCloser(bytes.NewBuffer(respBody)) simply refactor your methods so you dont read the body twice in the same method